sources/LLARRAYELT

```diff
@@ -2130,6 +2120,49 @@ Copyright (c) 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992,
                                 "Return code to tell \ALLOCBLOCK.NEW to notice the new arrangement")
           (RETURN 0])
 )
+(DEFINEQ
+
+(BLOCKEQUALP
+  [LAMBDA (BLOCK1 BLOCK2)                                    (* ; "Edited 24-Dec-2023 21:14 by rmk")
+                                                             (* ; "Edited 10-Dec-2023 21:19 by rmk")
+
+    (* ;; "True if BLOCK1 and BLOCK2 are blocks (produced by \ALLOCBLOCK) of equal size and equivalent contents.  Small blocks are allocated as hunks. Hunks have their own datatypes, blocks have type NIL.  Either way the types have to be the same.")
+
+    (OR (EQ BLOCK1 BLOCK2)
+        (LET (HDR1 HDR2 NCELLS PTRS (TY (TYPENAME BLOCK1)))
+             (AND (EQ TY (TYPENAME BLOCK2))
+                  [if TY
+                      then                                   (* ; "Fix-sized hunks")
+                           (CL:WHEN (STRPOS "HUNK" TY)
+                               (SETQ PTRS (EQ (CHARCODE P)
+                                              (NTHCHARCODE TY 2)))
+                               [SETQ NCELLS (SUBATOM TY (ADD1 (CL:IF PTRS
+                                                                  (NCHARS "\PTRHUNK")
+                                                                  (NCHARS "\UNBOXEDHUNK"))])
```

Comment:
this seems to put off to runtime things that are constants (like NCHARS).
If someone made a datatype called FOOHUNK would the STRPOS succeed?
It's not clear what's going on here.  The comment gives a clue but what is the story?